R/input-check-helpers.R

```diff
@@ -0,0 +1,207 @@
+#' predictive samples, n (number of rows) being the number of data points and
+#' N (number of columns) the number of Monte Carlo samples
+#' @param type character, one of "continuous", "integer" or "binary" that
+#' defines the type of the forecast
+#' @param class character, either "vector" or "matrix" that determines the
+#' class the input has to correspond to
+#' @inheritParams ae_median_sample
+#' @return NULL
+#' @keywords internal
+
+check_predictions <- function(predictions,
+                              true_values = NULL,
+                              type = c("continuous", "integer", "binary"),
+                              class = c("vector", "matrix")) {
+  type <- match.arg(type)
+  class <- match.arg(class)
+
+  if (missing(predictions)) {
+    stop("argument 'predictions' missing")
+  }
+
+  if (class == "vector") {
+    if (!is.vector(predictions)) {
+      msg <- sprintf(
+        "'predictions' should be a vector. Instead `%s` was found",
+        class(predictions[1])
+      )
+      stop(msg)
+    }
+    if (!is.null(true_values) && length(predictions) != length(true_values)) {
+      msg <- sprintf(
+        "Mismatch: 'true_values' has length `%s`, but 'predictions' has length `%s`.", # nolint
+        length(true_values), length(predictions)
+      )
+      stop(msg)
+    }
+  }
+
+  if (class == "matrix") {
+    if (!is.matrix(predictions)) {
+      msg <- sprintf(
+        "'predictions' should be a matrix. Instead `%s` was found",
+        class(predictions[1])
+      )
+      stop(msg)
+    }
+    if (!is.null(true_values) && nrow(predictions) != length(true_values)) {
+      msg <- sprintf(
+        "Mismatch: 'true_values' has length `%s`, but 'predictions' has `%s` rows.",
+        length(true_values), nrow(predictions)
+      )
+      stop(msg)
+    }
+  }
+
+  if (type == "integer") {
+    if (isFALSE(all.equal(predictions, as.integer(predictions)))) {
+      warning("Prediction type should be 'integer', but some of the predictions are not integers")
+    }
+  }
+
+  if (type == "binary") {
+    if (isFALSE(all(predictions >= 0) & all(predictions <= 1))) {
+      stop("For a binary forecast, all predictions should be probabilities between 0 or 1.")
+    }
+  }
+
+  return(NULL)
+}
+
+
+#' @title Check Observed Value Input For Lower-level Scoring Functions
+#'
+#' @description
+#' Helper function to check inputs for lower-level score functions.
+#' @inheritParams check_predictions
+#' @return NULL
+#' @keywords internal
+
+check_true_values <- function(true_values,
+                              type = c("continuous", "integer", "binary")) {
+  type <- match.arg(type)
+  if (missing(true_values)) {
+    stop("true_values argument is missing")
+  }
+
+  if (type == "integer") {
+    if (isFALSE(all.equal(true_values, as.integer(true_values)))) {
+      stop("Some of the true_values are not integers")
+    }
+  }
+
+  if (type == "binary") {
+    if (isFALSE(all(true_values %in% c(0, 1)))) {
```

Comment:
`all()` (as opposed to `all.equal()`) always returns `TRUE` or `FALSE` so this is not required.

But `all.equal()` might be better here to deal with floating point precision errors.